src/System.Memory/src/System/Memory.cs

```diff
@@ -143,9 +149,7 @@ public Memory<T> Slice(int start)
             if ((uint)start > (uint)_length)
                 ThrowHelper.ThrowArgumentOutOfRangeException(ExceptionArgument.start);
 
-            if (_index < 0)
-                return new Memory<T>((OwnedMemory<T>)_arrayOrOwnedMemory, (_index & RemoveOwnedFlagBitMask) + start, _length - start);
-            return new Memory<T>((T[])_arrayOrOwnedMemory, _index + start, _length - start);
+            return new Memory<T>(_object, _index + start, _length - start);
```

Comment:
If _index is a negative number (i.e. you get Memory from OwnedMemory.Memory), the slice operation won't work as expected. Don't we need to `RemoveOwnedFlagBitMask`?

Reply:
It does.  Let's say _index is 23, but it's from owned memory so it's 23 | 0x80000000.  We then increment it by a start of 3, so (23 | 0x80000000) + 3.  And then we strip the bits off of that.  ((23 | 0x80000000) + 3) & 0x7FFFFFFF.  That's 26.

Is there a specific case you're concerned about?

Reply:
> Is there a specific case you're concerned about?

Nope. That makes sense now. Thanks for the explanation.
I tried to reason about the boundary cases (near int.MaxValue) and convinced myself.

I was wrongfully working under the assumption that we are negating the number, i.e. `(23 | 0x80000000) == -23` when it is actually `int.MinValue + 23 == -2147483625`